structure/dependencies.go

```diff
@@ -116,20 +115,24 @@ func (t dependencyTree) ExpandDependencies(set PackageSet) {
 	}
 
 	changed := true
+	visited := NewPackageSet()
 	for changed {
 		changed = false
+
 		for _, pkg := range set.Enumerate() {
-			if deps, ok := t.m[pkg]; ok {
-				for dep := range deps {
-					if !set.Contains(dep) {
-						log.Debugf("package %s has package %s as a dependency, expanding tree", pkg, dep)
-						changed = true
-						set.Add(dep)
-					}
+			deps, ok := t.m[pkg]
+			if !ok {
```

Comment:
@jacopoRufini ma qui quando fai `continue`, `changed` è settato a `false` quindi in realtà stai facendo implicitamente `break`, giusto?

Reply:
ah no, scusa, so scemo io, stai facendo continue nel loop più interno